charts/opentelemetry-kube-stack/values.yaml

```diff
@@ -341,3 +341,173 @@ clusterRole:
 
   # Rules for the cluster role
   rules: []
+
+# Instrumentation configuration
+instrumentation:
+  # Whether instrumentation is enabled or not
+  enabled: false
+  labels: {}
+  annotations: {}
+
+  # Exporter configuration
+  exporter:
+    # This is the default collector's service
+    # Upon creation of a tracing collector, edit this endpoint.
+    endpoint: http://collector-collector:4317
+
+  # Resource configuration
+  resource:
+    resourceAttributes: {}
+    # environment: dev
+    addK8sUIDAttributes: true
+
+  # Propagators configuration
+  propagators:
+    - tracecontext
+    - baggage
+    - b3
+    - b3multi
+    - jaeger
+    - xray
+    - ottrace
+
+  # Sampler configuration
+  sampler:
+    type: parentbased_always_on
+    argument: "0.25"
+
+  # Environment variables for instrumentation
+  env: []
+  # - name: ENV_VAR1
+  #   value: value1
+  # - name: ENV_VAR2
+  #   value: value2
+
+  # Java agent configuration
+  java: {}
```

Comment:
A chart project I work on ([link](https://github.com/signalfx/splunk-otel-collector-chart/blob/eb8a3285a8d8707de242509baafb7f372643336e/helm-charts/splunk-otel-collector/values.yaml#L1197)) that integrates with the  collector, operator, and opentelemetry.io/v1alpha1 instrumentation as part of our solution specifically uses pinned tag values that are not "latest". 
  - We found it important to explicitly set and document the versions of instrumentation libraries used. This aids in customer understanding and records the tested versions among components.
  - The operator project maintains a list of instrumentation library versions ([versions.txt](https://github.com/open-telemetry/opentelemetry-operator/blob/main/versions.txt)) that tracks values per operator release, this information can be useful to know what versions were tested with the involved components.
  - While users have the option to specify their instrumentation versions, I suggest this chart should define the specific default versions used for instrumentations libraries to avoid breaking changes in environments. I would suggest the default values match the versions of components that were tested to work together.

Reply:
The current PR might be good enough as is for this topic. It might be worth creating a follow up issue for the work to ensure the used instrumentation version values are in sync with the version of the operator used. If a new ticket is created, I would like to volunteer to work on it.

Reply:
yeah! this makes a lot of sense. I would very much appreciate a hand with that as well. I can make the ticket.

Reply:
https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1129#issue-2233785704